engine/src/test/java/org/camunda/bpm/engine/test/bpmn/gateway/ParallelGatewayTest.java

```diff
@@ -373,4 +375,41 @@ public void testExplicitParallelGatewayAfterSignalBehavior() {
     // then
     assertEquals(3, taskService.createTaskQuery().count());
   }
+
+  @Deployment
+  @Test
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_FULL)
+  public void testParallelGatewayCancellationHistoryEvent() {
+    // given
+    // a process with one splitting and one merging parallel gateway and two parallel sequence flows between them
+    // one sequence flow has a wait state "Event_Wait", the other has none
+    var pi = runtimeService.startProcessInstanceByKey("parallelProcess");
+
+    var currentTime = new Date(5000);
+    ClockUtil.setCurrentTime(currentTime);
+
+    // when
+    // cancel "Event_Wait" first
+    // then cancel "Gateway_out" (merging parallel gateway)
+    this.runtimeService.createProcessInstanceModification(pi.getId())
+        .cancelAllForActivity("Event_Wait")
+        .cancelAllForActivity("Gateway_out")
+        .execute();
+
+    // then
+    // the whole process instance is canceled and history is produced
+    assertThat(runtimeService.createProcessInstanceQuery().processInstanceId(pi.getId()).singleResult()).isNull();
+    assertThat(historyService.createHistoricProcessInstanceQuery().processInstanceId(pi.getId()).singleResult()).isNotNull();
+
+    //
+    var historicActivityInstanceEventWait = this.historyService.createHistoricActivityInstanceQuery()
+        .activityId("Event_Wait")
+        .singleResult();
+    assertThat(historicActivityInstanceEventWait.getEndTime()).isEqualToIgnoringMillis(currentTime);
```

Comment:
❓ Same as below/above comment:
```suggestion
    assertThat(historicActivityInstanceEventWait.getEndTime()).isCloseTo(currentTime, 2000);
```